src/rev_ai/apiclient.py

```diff
@@ -28,51 +28,79 @@ def __init__(self, api_key):
             dashboard on rev.ai
         """
         if not api_key:
-            raise ValueError("API Key cannot be empty.")
+            raise ValueError('api_key must be provided')
+
         self.session = requests.Session()
         self.session.headers.update({
-            'Authorization': 'Bearer {api_key}'.format(api_key=api_key),
+            'Authorization': 'Bearer {}'.format(api_key),
             'User-Agent': 'python_sdk'
         })
 
-    def submit_job_url(self, media_url, options):
+    def submit_job_url(
+            self, media_url,
+            metadata=None,
+            callback_url=None,
+            skip_diarization=False):
         """Submit media given a URL for transcription.
         The audio data is downloaded from the URL.
 
         :param media_url: web location of the media file
-        :param options: JobSubmitOptions object for the job
+        :param metadata: info to associate with the transcription job
+        :param callback_url: callback url to invoke on job completion as a
+                             webhook
+        :param skip_diarization: should rev.ai skip diaization when
+                                 transcribing this file
         :returns: raw response data
         """
-        url_jobs = urljoin(self.base_url, "jobs")
-        payload = {
-            'media_url': media_url,
-            'metadata': options.metadata
-        }
-        if options.callback_url:
-            payload['callback_url'] = options.callback_url
+        if not media_url:
+            raise ValueError('media_url must be provided')
```

Comment:
this seems extra, why would the user provide a null argument to submit_job_url?

I don't think APIs should have excess safety logic. IMO raw as-close-to-the-metal is preferrable

Reply:
I think it's a bit arbitrary to just check falesy values, you could go further and also validate URL schema and stuff like that. But I can see your point.